typescript/core/src/McapIndexedReader.ts

```diff
@@ -51,6 +54,9 @@ export class McapIndexedReader {
     this.summaryOffsetsByOpcode = args.summaryOffsetsByOpcode;
     this.header = args.header;
     this.footer = args.footer;
+    if (args.dataEndOffset != undefined) {
```

Comment:
nit: probably don't need the `if` around this, since the property is already optional